File: 2026/06-June/07-Sunday/solution.js
var createBinaryTree = function(descriptions) {
    const nodes = new Map();
    const children = new Set();

    for (const [p, c, left] of descriptions) {
        if (!nodes.has(p))
            nodes.set(p, new TreeNode(p));

        if (!nodes.has(c))
            nodes.set(c, new TreeNode(c));

        if (left)
            nodes.get(p).left = nodes.get(c);
        else
            nodes.get(p).right = nodes.get(c);

        children.add(c);
    }

    for (const [p] of descriptions) {
        if (!children.has(p))
            return nodes.get(p);
    }

    return null;
};

File: 2026/06-June/07-Sunday/solution.go
func createBinaryTree(descriptions [][]int) *TreeNode {
    nodes := map[int]*TreeNode{}
    children := map[int]bool{}

    for _, d := range descriptions {
        p, c, left := d[0], d[1], d[2]

        if nodes[p] == nil {
            nodes[p] = &TreeNode{Val: p}
        }

        if nodes[c] == nil {
            nodes[c] = &TreeNode{Val: c}
        }

        if left == 1 {
            nodes[p].Left = nodes[c]
        } else {
            nodes[p].Right = nodes[c]
        }

        children[c] = true
    }

    for _, d := range descriptions {
        if !children[d[0]] {
            return nodes[d[0]]
        }
    }

    return nil
}
